hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -230,18 +301,17 @@ def append_int(n_bits, value):
                     return bytes(novel_prefix)
                 branch = current_node.transition
                 assert isinstance(branch, Branch)
-                n_bits = branch.bit_length
 
                 check_counter = 0
                 while True:
-                    k = random.getrandbits(n_bits)
+                    (v, buf) = draw(branch.ir_type, branch.kwargs)
```

Comment:
implicitly using the random of `ConjectureData(max_length=BUFFER_SIZE, prefix=b"", random=random)` here to draw a random new value. We can't use `k = random.getrandbits(n_bits)` as before because the value of the child is no longer guaranteed to be an integer.